File: app/src/main/java/com/elementary/tasks/core/SplashScreen.java
package com.elementary.tasks.core;

import android.content.Intent;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.os.Bundle;

import com.elementary.tasks.core.async.EnableThread;
import com.elementary.tasks.core.data.AppDb;
import com.elementary.tasks.core.services.PermanentReminderReceiver;
import com.elementary.tasks.core.utils.Notifier;
import com.elementary.tasks.core.utils.Prefs;
import com.elementary.tasks.groups.GroupsUtil;
import com.elementary.tasks.intro.IntroActivity;
import com.elementary.tasks.navigation.MainActivity;

import androidx.annotation.Nullable;

public class SplashScreen extends ThemedActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initPrefs();
        if (Prefs.getInstance(this).isSbNotificationEnabled()) {
            Notifier.updateReminderPermanent(this, PermanentReminderReceiver.ACTION_SHOW);
        }
    }

    private void checkIfAppUpdated() {
        try {
            PackageInfo info = getPackageManager().getPackageInfo(getPackageName(), 0);
            if (!Prefs.getInstance(this).getVersion(info.versionName)) {
                Prefs.getInstance(this).saveVersionBoolean(info.versionName);
                new EnableThread(this).start();
            }
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        gotoApp();
    }

    private void gotoApp() {
        checkIfAppUpdated();
        if (!Prefs.getInstance(this).isUserLogged()) {
            openIntroScreen();
        } else {
            initGroups();
            runApplication();
        }
    }

    private void openIntroScreen() {
        startActivity(new Intent(this, IntroActivity.class));
        finish();
    }

    private void initGroups() {
        if (AppDb.getAppDatabase(this).groupDao().getAll().size() == 0) {
            GroupsUtil.initDefault(this);
        }
    }

    private void initPrefs() {
        Prefs prefs = Prefs.getInstance(this);
        prefs.initPrefs(this);
        prefs.checkPrefs();
    }

    private void runApplication() {
        startActivity(new Intent(this, MainActivity.class));
        finish();
    }
}

File: app/src/main/java/com/elementary/tasks/core/SplashScreen.kt
package com.elementary.tasks.core

import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import com.elementary.tasks.core.data.AppDb
import com.elementary.tasks.core.services.PermanentReminderReceiver
import com.elementary.tasks.core.utils.EnableThread
import com.elementary.tasks.core.utils.Module
import com.elementary.tasks.databinding.ActivityLoginBinding
import com.elementary.tasks.groups.GroupsUtil
import com.elementary.tasks.login.LoginActivity
import com.elementary.tasks.navigation.MainActivity
import com.elementary.tasks.navigation.settings.security.PinLoginActivity

class SplashScreen : ThemedActivity<ActivityLoginBinding>() {

    override fun layoutRes(): Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initPrefs()
        if (prefs.isSbNotificationEnabled) {
            notifier.updateReminderPermanent(PermanentReminderReceiver.ACTION_SHOW)
        }
    }

    override fun applyTheme(): Boolean = false

    private fun checkIfAppUpdated() {
        try {
            val info = packageManager.getPackageInfo(packageName, 0)
            if (!prefs.getVersion(info.versionName)) {
                prefs.saveVersionBoolean(info.versionName)
                EnableThread.run(this)
            }
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
        }
    }

    override fun onResume() {
        super.onResume()
        gotoApp()
    }

    private fun gotoApp() {
        checkIfAppUpdated()
        if (!prefs.isUserLogged) {
            openIntroScreen()
        } else {
            initGroups()
            if (prefs.hasPinCode) {
                openPinLogin()
            } else {
                runApplication()
            }
        }
    }

    private fun openPinLogin() {
        startActivity(Intent(this, PinLoginActivity::class.java))
        finish()
    }

    private fun openIntroScreen() {
        Module.checkComponents(this)
        startActivity(Intent(this, LoginActivity::class.java))
        finish()
    }

    private fun initGroups() {
        if (AppDb.getAppDatabase(this).reminderGroupDao().all().isEmpty()) {
            GroupsUtil.initDefault(this)
        }
    }

    private fun initPrefs() {
        prefs.initPrefs(this)
        prefs.checkPrefs()
    }

    private fun runApplication() {
        startActivity(Intent(this, MainActivity::class.java))
        finish()
    }
}